src/vm/virtualMachine.ts

```diff
@@ -59,7 +67,7 @@ export class VirtualMachine {
     if (dstError !== undefined) {
       return { value: undefined, error: dstError };
     }
-    const dst = this.segments.memory.get(dstAddr);
+    let dst = this.segments.memory.get(dstAddr);
```

Comment:
we need to document in which cases dst is already defined in memory, as opposed to it being undefined, where we'd need to deduce it